plugins/Toolbox/src/Toolbox.py

```diff
@@ -0,0 +1,609 @@
+            #                 return
+            #
+            #             # Create model and apply metadata:
+            #             if not self._models[type]:
+            #                 Logger.log("e", "Could not find the %s model.", type)
+            #                 break
+            #             self._metadata[type] = json_data["data"]
+            #             self._models[type].setMetadata(self._metadata[type])
+            #             self.metadataChanged.emit()
+            #             self.setViewPage("overview")
+            #             return
+            #         except json.decoder.JSONDecodeError:
+            #             Logger.log("w", "Toolbox: Received invalid JSON for %s.", type)
+            #             break
+
+            if reply.url() == self._request_urls["packages"]:
+                try:
+                    json_data = json.loads(bytes(reply.readAll()).decode("utf-8"))
+
+                    # Check for errors:
+                    if "errors" in json_data:
+                        for error in json_data["errors"]:
+                            Logger.log("e", "%s", error["title"])
+                        return
+
+                    # Create packages model with all packages:
+                    if not self._models["packages"]:
+                        self._models["packages"] = PackagesModel(self)
+                    self._metadata["packages"] = json_data["data"]
+                    self._models["packages"].setMetadata(self._metadata["packages"])
+                    self.metadataChanged.emit()
+
+                    # Create authors model with all authors:
+                    if not self._models["authors"]:
+                        self._models["authors"] = AuthorsModel()
+                    # TODO: Replace this with a proper API call:
+                    for package in self._metadata["packages"]:
+                        if package["author"] not in self._metadata["authors"]:
+                            self._metadata["authors"].append(package["author"])
+
+                    for author in self._metadata["authors"]:
+                        if "package_count" not in author:
+                            author["package_count"] = 0
+
+                        for package in self._metadata["packages"]:
+                            if package["author"]["name"] == author["name"]:
+                                author["package_count"] += 1
+                                author["type"] = package["package_type"]
+                                if "icon_url" in package:
+                                    author["icon_url"] = package["icon_url"]
+
+                    self._models["authors"].setMetadata(self._metadata["authors"])
+                    self.metadataChanged.emit()
+
+                    if not self._models["materials_showcase"]:
+                        self._models["materials_showcase"] = AuthorsModel(self)
```

Comment:
The `AuthorsModel` is always constructed in the `__init__` function, so it will never hit this `if` statement.

Reply:
Yup. Artifact from before I decided "why not just create it on init?"